### ISSUE TITLE
Switch to BouncyCastle by default

### ISSUE THREAD

**Issue by mikera** [Convex-Dev/convex#410]
Benchmarks by @helins suggest BouncyCastle is faster than LazySodium right now.

We should switch to BC for now, until a better option is available.

**Comment by helins** [comment]
Indeed, but let me first reproduce it on CI across a matrix OS x JVM version :)

**Comment by helins** [comment]
Wow, good thing I tripled-check everything and took some time to ensure it was done right.

CI revealed the opposite: LazySodium is up to 2x as fast as BouncyCastle!
Confirmed on a matrix `[macos-x86, linux ]` x `java[11, 17, 18]`.

The only exception is precisely Java 18 running on my M1 chip. Which is what I was doing initially because I needed access to Project Panama. Something must be amiss with JNA in that particular configuration. But even on machine, LS is about 1.75x faster than BC when running Java 17.

The good news is that signature validation seems to take about 40 micros on decent hardware (even CI runners perform at 70 micros which is already lower than I initially expected).

---

So the only reason for supporting a BC backend would be clients that need a pure Java implementation. Currently, this is only the Convex Shell. Still pretty important WRT to dev tooling though.
